controllers/registrationManagement.js

```diff
@@ -0,0 +1,82 @@
+const registrationManagementRouter = require('express').Router()
+const db = require('../models/index')
+const checkAdmin = require('../utils/middleware/routeChecks').checkAdmin
+
+const handleDatabaseError = (res, error) => {
+  console.log(error)
+  res.status(500).json({ error: 'database error' })
+}
+const create = (req, res) => {
+  const { registrationManagement } = req.body
+
+  db.RegistrationManagement.create(registrationManagement)
+    .then((createdRegistrationManagement) => {
+      res.status(201).json(createdRegistrationManagement)
+    })
+    .catch((error) => {
+      handleDatabaseError(res, error)
+    })
+}
+
+const isDefined = (attribute) => {
+  return attribute !== undefined
+}
+
+const createChecks = (req, res) => {
+  const {
+    project_registration_open,
+    project_registration_message,
+    topic_registration_open,
+    topic_registration_message
+  } = req.body.registrationManagement
+
+  if (
+    !isDefined(project_registration_open) ||
+    !isDefined(project_registration_message) ||
+    !isDefined(topic_registration_open) ||
+    !isDefined(topic_registration_message)
+  ) {
+    res.status(400).json({
+      error: 'All attributes must be defined'
+    })
+  } else if (
+    !project_registration_open &&
+    project_registration_message.length === 0
+  ) {
+    res.status(400).json({
+      error: 'Message must be provided when project registration is closed'
+    })
+  } else if (
+    !topic_registration_open &&
+    topic_registration_message.length === 0
+  ) {
+    res.status(400).json({
+      error: 'Message must be provided when topic registration is closed'
+    })
+  } else {
```

Comment:
Could maybe move everything above this to a `validateBody` function? Or at least return early with `return res.status(400).json({ error: ... })`. Express doesn't use the return values so really it's the same as doing 
```
res.status(400).json({ error: ... })
return
```
I think that looks a bit nicer:

```javascript
const isNil = (value) => value === undefined || value === null

const validateRegistrationManagement = (registrationManagement) => {
  if (!registrationManagement) {
    return 'All attributes must be defined'
  }

  const {
    project_registration_open,
    project_registration_message,
    topic_registration_open,
    topic_registration_message
  } = registrationManagement

  if (
    isNil(project_registration_open) ||
    isNil(project_registration_message) ||
    isNil(topic_registration_open) ||
    isNil(topic_registration_message)
  ) {
    return 'All attributes must be defined'
  }

  if (!project_registration_open && project_registration_message.length === 0) {
    return 'Message must be provided when project registration is closed'
  }

  if (!topic_registration_open && topic_registration_message.length === 0) {
    return 'Message must be provided when topic registration is closed'
  }

  return null
}

registrationManagementRouter.post('/', checkAdmin, async (req, res) => {
  const { registrationManagement } = req.body

  const error = validateRegistrationManagement(registrationManagement)
  if (error) {
    return res.status(400).json({ error })
  }

  create(req, res)
})
```